mmdet/core/anchor/point_generator.py

```diff
@@ -140,14 +149,21 @@ def single_level_grid_priors(self,
         stride_w, stride_h = self.strides[level_idx]
         shift_x = (torch.arange(0., feat_w, device=device) +
                    self.offset) * stride_w
+        # keep featmap_size as Tensor instead of int, so that we
+        # can covert to ONNX correctly
+        shift_x = shift_x.to(dtype)
+
         shift_y = (torch.arange(0., feat_h, device=device) +
                    self.offset) * stride_h
+        # keep featmap_size as Tensor instead of int, so that we
+        # can covert to ONNX correctly
+        shift_y = shift_y.to(dtype)
         shift_xx, shift_yy = self._meshgrid(shift_x, shift_y)
         if not with_stride:
             shifts = torch.stack([shift_xx, shift_yy], dim=-1)
         else:
-            stride_w = shift_xx.new_full((len(shift_xx), ), stride_w)
-            stride_h = shift_xx.new_full((len(shift_yy), ), stride_h)
+            stride_w = shift_xx.new_full((len(shift_xx), ), stride_w).to(dtype)
```

Comment:
new_full should also accept dtype

Reply:
Please @RunningLeon take a look

Reply:
if `shift_xx` is a tensor, could use `shift_xx.shape[0]` insread of `len(shift_xx)`